mani_skill/trajectory/merge_trajectory.py

```diff
@@ -1,60 +1,76 @@
+    """
+    logger.info(f"Merging {output_path}")
 
     merged_h5_file = h5py.File(output_path, "w")
     merged_json_path = output_path.replace(".h5", ".json")
-    merged_json_data = {"env_info": {}, "episodes": []}
-    _env_info = None
+    merged_json_data = {"episodes": []}
     cnt = 0
 
     for traj_path in traj_paths:
         traj_path = str(traj_path)
-        print("Merging", traj_path)
-
-        h5_file = h5py.File(traj_path, "r")
-        json_path = traj_path.replace(".h5", ".json")
-        json_data = load_json(json_path)
-
-        # Check env info
-        env_info = json_data["env_info"]
-        if _env_info is None:
-            _env_info = env_info
-            merged_json_data["env_info"] = _env_info
-        else:
-            assert str(env_info) == str(_env_info), traj_path
-
-        # Merge
-        for ep in json_data["episodes"]:
-            episode_id = ep["episode_id"]
-            traj_id = f"traj_{episode_id}"
-
-            # Copy h5 data
-            if recompute_id:
-                new_traj_id = f"traj_{cnt}"
-            else:
-                new_traj_id = traj_id
-
-            assert new_traj_id not in merged_h5_file, new_traj_id
-            h5_file.copy(traj_id, merged_h5_file, new_traj_id)
-
-            # Copy json data
-            if recompute_id:
-                ep["episode_id"] = cnt
-            merged_json_data["episodes"].append(ep)
-
-            cnt += 1
-
-        h5_file.close()
+        logger.info(f"Merging{traj_path}")
+
+        with h5py.File(traj_path, "r") as h5_file:
+            json_data = load_json(traj_path.replace(".h5", ".json"))
+            
+            # For keys other than episodes, keep the first data
+            # and check if there is any conflict with other data.
+            for key, value in json_data.items():
+                if key == "episodes":
+                    continue
+                if key not in merged_json_data:
+                    merged_json_data[key] = value
+                else:
+                    if merged_json_data[key] != value:
+                        logger.warning(f"Conflict detected for key {key} in {traj_path}: {merged_json_data[key]} != {value}")
+
+            # Merge episodes
+            for ep in json_data["episodes"]:
+                episode_id = ep["episode_id"]
+                traj_id = f"traj_{episode_id}"
+
+                # Copy h5 data
+                if recompute_id:
+                    new_traj_id = f"traj_{cnt}"
+                else:
+                    new_traj_id = traj_id
+
+                assert new_traj_id not in merged_h5_file, new_traj_id
+                h5_file.copy(traj_id, merged_h5_file, new_traj_id)
+
+                # Copy json data
+                if recompute_id:
+                    ep["episode_id"] = cnt
+                merged_json_data["episodes"].append(ep)
+
+                cnt += 1
 
-    # Ignore commit info
```

Comment:
Commit info is now preserved in the trajectory data because I thought it could be useful when the version of trajectory generation code is different (e.g., if someone made changes on panda motionplanner).